### PR TITLE
Add BEP1&2 proposals

### PR DESCRIPTION
**Pull request by huangsuyu**
In this PR, BEP 1&2 were added.
BEP1 is the general guideline and BEP2 is about tokens on Binance Chain.

### CODE REVIEW

File: BEP1.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  * [6.  License](#6--license)
+
+
+## 1.  What is BEP?
+
+BEP stands for Binance Chain Evolution Proposal. Each BEP will be a proposal document providing information to the Binance Chain/DEX community, or describing a new feature for Binance/DEX or its improvements. The BEP should provide a concise technical specification of the feature or improvement and the rationale behind it. Each BEP proposer is responsible for building consensus within the community and documenting dissenting opinions. Each BEP has a unique index number.
+
+## 2.  BEP Rationale
+
+BEP is the primary mechanism for proposing new features, for collecting community technical input on an issue, and for documenting the design decisions that have gone into Binance Chain. Because the BEP are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
```
Comment (darren-liu) [inline BEP1.md]: "documenting the design decisions that go into Binance Chain."

File: BEP1.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+For Binance Chain contributors, BEPs are a convenient way to track the progress of their implementation. This will give end users a convenient way to know the current status of a given feature, function or improvement.
+
+##  3.  BEP Types
+
+There are three types of BEP:
+
+- Feature Request: A Feature Request BEP describes changes on Binance Chain or/and Binance DEX, such as a change to the network protocol, proposer selection mechanism in consensus algorithm, change in block size or fee mechanism in application level.
```
Comment (darren-liu) [inline BEP1.md]: "describes functional changes on Binance"

File: BEP1.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+- Improvement: Improvements are advice gathered from the community.
+- Standard: This kind of proposal will change the workflow of Binance Chain working process, like BEP itself.
```
Comment (darren-liu) [inline BEP1.md]: "like this BEP itself"

File: BEP2.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+## 3.  Motivation
+
+Design and issue asset on the Binance Chain, as the basic economic foundations of the blockchain.
+
+## 4.  Specification
+
+### 4.1  Native Token on Binance Chain: BNB
+
+The Binance Token, BNB, is the native asset on Binance Chain and created within Genesis Block. BNB was an ERC20 token, but after Binance Chain is live, all BNB ERC20 tokens are swapped for BNB token on Binance Chain. All users who hold BNB ERC20 tokens can deposit them to Binance.com, and upon withdrawal, the new Binance Chain native tokens will be sent to their new addresses.
```
Comment (darren-liu) [inline BEP2.md]: add as 2nd sentence: `As the native asset, BNB would be used for fees (gas) and staking on Binance Chain.`

File: BEP2.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+### 4.2 Token Properties
+
+- Source Address: Source Address is the owner of the issued token.
+
+- Token Name: Token Name represents the long name of the token - e.g. "MyToken". 
+
+- Symbol: Symbol is the identifier of the newly issued token.
```
Comment (darren-liu) [inline BEP2.md]: add a new line break here?

**Comment by wjmelements** [comment]
Why does BIP2 use signed integers for amount and supply?

File: BEP2.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+**Data Structure for Issue Operation**: A data structure is needed to represent the new token:
+
+| **Field**    | **Type** | **Description**                                              |
+| :------------ | :-------- | :------------------------------------------------------------ |
+| Name         | string   | Name of the newly issued asset, limited to 32 unicode characters,  e.g. "ABCcoin" |
+| Symbol       | string   | The length of the string for representing this asset is between 3 and 8 alphanumeric characters and is case insensitive. ".B" suffixed symbol is also allowed for pegging to those tokens already exist on other chains. The symbol is suffixed with the first 3 bytes of the issue transaction hash to remove a constraint of requiring unique token names. The native token, BNB, does not require this suffix. |
+| Total Supply | int64    | The total supply for this token can have a maximum of 8 digits of decimal and is boosted by 1e8 in order to store as int64. The amount before boosting should not exceed 90 billion. |
```
Comment (wjmelements) [inline BEP2.md]: Why is there an artificial supply cap?

Comment (wjmelements) [reply BEP2.md]: Ah, the cap is similar to the amount possible when subdividing by 1e8. The real limit is ~92.23372 Billion.

**Comment by wjmelements** [comment]
You should probably increase the number of bits to 128 or 256 from 63.

**Comment by wjmelements** [comment]
The supply constraint is crippling. Tether could not fit on Binance chain today without issuing multiple tokens because the limit is $900 Million. You may think it is currently $90 Billion, but because burns must be processed in fixed units of 1e8, the actual token must represent the smallest denomination of currency. For the dollar this is one cent: $0.01. So, less than one-third of Tether's supply can fit on Binance Chain.

**Comment by wjmelements** [comment]
https://github.com/binance-chain/BEPs/pull/4

**Review by jonesk7734 — APPROVED** [review]
😀

**Review by Yedfree — APPROVED** [review]

**Pull request merged**